### DISCUSSION TITLE
往一个主键的auto_id=False的collection中重复insert相同主键的数据是否会占用更多的空间？

### DISCUSSION THREAD

**Discussion by ArcturusMensk** [milvus-io/milvus#25411] [Q&A and General discussion] [answered]
def create_test_collection():
   utility.drop_collection('test_collection')
   fields = [
       FieldSchema(name='doc_id', dtype=DataType.INT64, is_primary=True, auto_id=False),
       FieldSchema(name='embedding', dtype=DataType.FLOAT_VECTOR, dim=4),
        FieldSchema(name='desc', dtype=DataType.VARCHAR, max_length=100)
    ]
    schema = CollectionSchema(fields=fields, dscription='test')
    collection = Collection(name='test_collection', schema=schema)
    index_params = {'metric_type': "IP", "index_type": "IVF_FLAT", "params": {"nlist": 1024}}
    collection.create_index(field_name='embedding', index_params=index_params)
    return collection

test_collection = create_test_collection()
test_insert_data = [{'doc_id': i, 'embedding': [i+3/10, i+4/10, i+5/10, i+6/10], 'desc': f"{i}{i}"} for i in range(1, 10)]

test_collection.load()
test_collection.insert(test_insert_data)
test_collection.flush()
test_collection.num_entities
-- output: 9

test_collection.insert(test_insert_data)
test_collection.flush()
test_collection.num_entities
-- output: 18

pymilvus == 2.2.12

**Comment by xiaofan-luan** [comment]
是的，milvus目前没有主键去重能力

**Comment by xiaofan-luan** [comment, marked as answer]
重复主键compaction的时候会被清理，但不保证何时发生。
